HPXMLtoOpenStudio/resources/hpxml_defaults.rb

```diff
@@ -1520,7 +1520,9 @@ def self.apply_detailed_performance_data_for_var_speed_systems(hpxml)
     hpxml.heat_pumps.each do |heat_pump|
       next unless [HPXML::HVACTypeHeatPumpAirToAir,
                    HPXML::HVACTypeHeatPumpMiniSplit].include? heat_pump.heat_pump_type
-
+      # Assign minimum capacity here to avoid E+ simulation error
+      min_capacity = 1.0
+      heat_pump.heating_capacity = [heat_pump.heating_capacity, min_capacity].max
```

Comment:
This will execute before writing in.xml, so our minimum value will be reflected on the in.xml. Couldn't come up with a better approach off the top of my head. I tried applying minimum value for each data point later at the `ensure_nonzero_sizing_values`, it will still catch some E+ errors saying that the maximum speed airflow should be greater than minimum speed. Will leave it here to try to pass ci first

Reply:
@yzhou601 Can you give me an example of how to reproduce the E+ error (assuming I comment out this code)? I can take a closer look at whether there's a better way to address this if I understand the situation better.

Reply:
[in.idf.txt](https://github.com/NREL/OpenStudio-HPXML/files/12296569/in.idf.txt)
[eplusout.err.txt](https://github.com/NREL/OpenStudio-HPXML/files/12296570/eplusout.err.txt)
See attached the E+ error by running .\workflow\sample_files\base-hvac-mini-split-heat-pump-ducted-cooling-only.xml
The code snippet is like adding the following lines to the ensure_non_zero_sizing_values:

>       if not hp_sys.heating_detailed_performance_data.empty?
>         hp_sys.heating_detailed_performance_data.each do |dp|
>           dp.capacity = [dp.capacity, min_capacity].max
>         end
>       end
>       if not hp_sys.cooling_detailed_performance_data.empty?
>         hp_sys.cooling_detailed_performance_data.each do |dp|
>           dp.capacity = [dp.capacity, min_capacity].max
>         end
>       end

The warnings thrown by E+ is complaining the small airflow.

Reply:
What do you think about this? It increments the capacity by speed rather than setting them all to the same minimum capacity. It seems to work for this test file at least.

```
      if not hp_sys.heating_detailed_performance_data.empty?
        hp_sys.heating_detailed_performance_data.each do |dp|
          speed = dp.capacity_description == 'minimum' ? 1 : 2
          dp.capacity = [dp.capacity, min_capacity * speed].max
        end
      end
      if not hp_sys.cooling_detailed_performance_data.empty?
        hp_sys.cooling_detailed_performance_data.each do |dp|
          speed = dp.capacity_description == 'minimum' ? 1 : 2
          dp.capacity = [dp.capacity, min_capacity * speed].max
        end
      end
```